federationtypes.go

```diff
@@ -305,51 +309,10 @@ type RespMakeJoin struct {
 
 // A RespSendJoin is the content of a response to PUT /_matrix/federation/v1/send_join/{roomID}/{eventID}
 // It has the same data as a response to /state, but in a slightly different wire format.
-type RespSendJoin RespState
-
-// MarshalJSON implements json.Marshaller
-func (r RespSendJoin) MarshalJSON() ([]byte, error) {
```

Comment:
These are still needed for now I think... They may just get removed at some point in the future.